ext/dom/parentnode.c

```diff
@@ -296,6 +319,11 @@ void dom_parent_node_prepend(dom_object *context, zval *nodes, int nodesc)
 		return;
 	}
 
+	if (UNEXPECTED(dom_hierarchy_node_list(parentNode, nodes, nodesc) != SUCCESS)) {
```

Comment:
Any reason for using ``!= SUCCESS``? Shouldn't really be an issue here but when I was converting a lot of ``int`` return values to ``zend_result`` this code pattern always made me triple check that I was not missing a case where something other than ``SUCCESS`` and ``FAILURE`` was returned.
```suggestion
	if (UNEXPECTED(dom_hierarchy_node_list(parentNode, nodes, nodesc) == FAILURE)) {
```

Reply:
I have a habit of doing `!= SUCCESS` because it's a comparison against 0, which encodes as a shorter instruction than a comparison with -1. I saw both `!= SUCCESS` and `== FAILURE` in the code so I just picked the one I liked most.
I can change it if you want?

Reply:
Either is fine frankly, I think I was told about the shorter instruction once.

I believe that one day ``zend_result`` can be a true alias for ``bool``... but that's a lot of code auditing.